pkg/crypto/tinkcrypto/primitive/jwe/ecdhes/ecdhes.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package ecdhes provides implementations of JWE encryption using ECDH-ES KW primitives.
+//
+// The functionality of ecdhes Encryption is represented as a pair of
+// primitives (interfaces):
+//
+//  * ECDHESEncrypt for encryption of data as JWE
+//
+//  * ECDHESDecrypt for decryption of JWE data and returning decrypted plaintext
+//
+//
+// Example:
+//
+//   package main
+//
+//   import (
+//       "github.com/google/tink/go/core/registry"
+//       "github.com/google/tink/go/keyset"
+//
+//       "github.com/aries-framework-go/pkg/crypto/tinkcrypto/jwe/ecdhes"
+//   )
+//
+//   func main() {
+//
+//       kh , err := keyset.NewHandle(ecdhes.AES256GCMECDHESKeyTemplate())
+//       if err != nil {
+//           //handle error
+//       }
+//
+//       pubKH, err := kh.Public()
+//       if err != nil {
+//           //handle error
+//       }
+//
+//       e := ecdhes.NewECDHESEncrypt(pubKH)
+//
+//       ct, err = e.Encrypt([]byte("secret message"), []byte("some aad"))
+//       if err != nil {
+//           // handle error
+//       }
+//
+//       khd , err := keyset.NewHandle( .....); // get a handle on the decryption key material for a recipient
+//       d := ecdhes.NewECDHESDecrypt(khd)
+//
+//       pt, err := d.Decrypt(ct)
+//       if err != nil {
+//           // handle error
+//       }
+//   }
+package ecdhes
+
+import (
+	"fmt"
+
+	"github.com/google/tink/go/core/registry"
+)
+
+// nolint: gochecknoinits
+func init() {
+	err := registry.RegisterKeyManager(newECDHESPrivateKeyManager())
```

Comment:
do we really need to use this registry singleton?

Reply:
I followed Tink's pattern with other primitives. They register key managers this way.

The registry is used to fetch primitives and their keys. We will need to change Tink to remove singleton.